docs/modules/ROOT/pages/adr/0043-appcat-framework-production-architecture.adoc

```diff
@@ -0,0 +1,1750 @@
+* *Composition Support*: Import system enables base configurations with distribution-specific overrides
+* *Override Pattern*: Supports defaults → service → distribution → profile composition chain
+* *Active Development*: Backed by Ant Group with growing community and ecosystem
+
+**Disadvantages:**
+
+* *Smaller Ecosystem*: Fewer tools and IDE support compared to more established languages like CUE
+* *Learning Curve*: Service Maintainers/Framework Engineers need to learn new syntax and constraint-based thinking
+* *Less Mature*: Newer language with potential for breaking changes as it evolves
+* *Limited Resources*: Fewer tutorials and examples
+
+===== Option B: CUE (Configure Unify Execute)
+
+**Description:**
+
+CUE is a data constraint language where types and values are unified.
+Configurations compose through constraint unification.
+Excels at validation but lacks native lambda functions - transformation logic must be expressed through constraint systems, making Helm value mappings verbose.
+
+**Advantages:**
+
+* *Unification Model*: Types and values unified - constraints automatically propagate and validate across compositions
+* *Exceptional Data Validation*: Catch configuration errors through mathematical constraint solving before any runtime execution
+* *Mature Tooling*: Language Server Protocol, IDE support, comprehensive CLI tools
+* *Constraint Propagation*: Automatically validates that all configuration layers satisfy constraints without explicit checks
+* *Export Flexibility*: Can export to multiple formats (JSON, YAML, Protocol Buffers) natively
+
+**Disadvantages:**
+
+* *Steeper Learning Curve*: Unification model requires different thinking - declarative constraints vs procedural logic
+* *Less Imperative*: Difficult to express procedural transformations like Helm value mapping with conditional logic
+* *No Native Lambda Functions*: Must encode logic as constraints
+* *Verbose Transformations*: Complex mapping functions require extensive boilerplate, reducing Service Maintainer velocity
+* *Limited Procedural Logic*: "If-then-else" patterns become constraint systems that are harder to read and maintain
+
+==== Decision
+
+*We use KCL.*
+
+==== Rationale
+
+1. *Lambda Functions*: Service Maintainers need to write Helm value mapping functions. KCL's lambda support makes this natural and concise.
+
+2. *Kubernetes-Native*: KCL is designed specifically for Kubernetes configuration, with built-in understanding of K8s resource structures.
+
+3. *Procedural Logic*: Helm value mapping often requires conditional logic. KCL handles this more naturally than CUE.
+
+4. *Simpler Mental Model*: For Service Maintainers and Framework Engineers, KCL's imperative style is closer to traditional programming, reducing learning curve.
+
+5. *Desired Pattern*: The composition function can load KCL-compiled ConfigMaps at runtime, providing clear separation between Framework (Go) and Service Maintainer (KCL).
```

Comment:
just a NIT: composition functions don't only support ConfigMaps inputs. It can be any object. It just has to look like a Kubernetes object.

We probably want our own datastructure to reflect the needs of the new framework.

Reply:
Updated the ADR to reference `AppCatServiceConfig` custom resources instead of a ConfigMap